### PR TITLE
Add next-best-action as a new Box

### PR DESCRIPTION
**Pull request by Botafogo1894**

### CODE REVIEW

**Comment by kzk** [comment]
Can I ask you to

- Add README.md
- Change the name from "NBA_metrics" to "next-best-action" for the consistency?

File: machine-learning-box/NBA_metrics/nba_activity_percentile_metrics.dig [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# error notification email
+_error:
+  mail>: body.txt
+  subject: this workflow failed
+  to: [dilyan.kovachev@treasure-data.com]
```
Comment (toru-takahashi) [inline machine-learning-box/NBA_metrics/nba_activity_percentile_metrics.dig]: Could you remove your email?

File: machine-learning-box/NBA_metrics/nba_activity_percentile_metrics.dig [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    user_master_join_key: td_client_id
+    timestamp_column: time_of_day
+    original_pageviews: pageviews
+    activity_table: pageviews_time_of_day
+    activity_table_join_key: td_client_id
+    enriched_user_master_table: user_master_enriched
+    enriched_user_master_table_temp: user_master_enriched_temp
+    time_source_column: activity_period
+    utm_source_column: td_url
+    cart_url: 'https://www.varidesk.com/cart'
```
Comment (toru-takahashi) [inline machine-learning-box/NBA_metrics/nba_activity_percentile_metrics.dig]: Could you change them to like https://www.ecsite-example.com/ ?

File: machine-learning-box/NBA_metrics/queries/cart_abandon.sql [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                CASE
+                  WHEN td_url = 'https://www.varidesk.com/cart' THEN 1
+                  ELSE 0
+                END
+              ) AS flag
+            FROM
+              pageviews A
+            WHERE
+              td_url IN(
+                'https://www.varidesk.com/cart',
```
Comment (toru-takahashi) [inline machine-learning-box/NBA_metrics/queries/cart_abandon.sql]: These should be variables.

File: machine-learning-box/NBA_metrics/queries/cart_abandon.sql [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+              ) OVER(
+                PARTITION BY td_client_id,
+                td_ip
+              ORDER BY
+                time
+              ) AS session_id,
+              time,
+              (
+                CASE
+                  WHEN td_url = 'https://www.varidesk.com/cart' THEN 1
```
Comment (toru-takahashi) [inline machine-learning-box/NBA_metrics/queries/cart_abandon.sql]: This should be a variable like ${cart_url}

Comment (Botafogo1894) [reply machine-learning-box/NBA_metrics/queries/cart_abandon.sql]: @toru-takahashi you should not be seeing any files from the NBA_metrics box at all. This is an older folder with old version of queries that I deleted from my branch on my computer. I am not sure why it still comes up during the pull request... :(   I need some help figuring out why you're seeing this folder.

File: machine-learning-box/NBA_metrics/nba_activity_percentile_metrics.dig (hunk 1 above)
Comment (Botafogo1894) [reply machine-learning-box/NBA_metrics/nba_activity_percentile_metrics.dig]: my e-mail should not be here in the final version of workflow. I think something strange is happening where you are not getting the latest version of my files from my "add_dilyan_kovachev" branch, but I am not sure how to fix this. Any suggestions?

**Pull request merged**